packages/gatsby-react-router-scroll/src/StateStorage.js

```diff
@@ -48,6 +48,6 @@ export default class SessionStorage {
   getStateKey(location, key) {
     const locationKey = location.key
     const stateKeyBase = `${STATE_KEY_PREFIX}${locationKey}`
-    return key == null ? stateKeyBase : `${stateKeyBase}|${key}`
+    return key === null ? stateKeyBase : `${stateKeyBase}|${key}`
```

Comment:
This changes the logic as it's not now checking for `undefined` https://stackoverflow.com/a/5515385/182702 If you want to use the strict equality check, could you also add a check for undefined?

Reply:
Sure, will do, thanks @KyleAMathews